mlflow/genai/scorers/base.py

```diff
@@ -295,10 +297,19 @@ def model_validate(cls, obj: Any) -> "Scorer":
 
     @classmethod
     def _reconstruct_decorator_scorer(cls, serialized: SerializedScorer) -> "Scorer":
-        """Reconstruct a decorator scorer from serialized data."""
         from mlflow.genai.scorers.scorer_utils import recreate_function
 
-        # Recreate the original function from source code
+        # NB: Custom (@scorer) scorers use exec() during deserialization, which poses a code
+        # execution risk. Only allow loading with Databricks tracking URIs where the execution
+        # environment is controlled.
```

Comment:
is it safe to load on my machine using a token?

Reply:
I ended up making it possible to register if using a databricks uri (this is safe and supports remote development) but only allowed loading if within a databricks environment. Added NB comments reflecting why the disparity.